nixos/outline: shellcheck errors

### Nixpkgs version

- Stable (25.05)

### Describe the bug

When I enable shellcheck for the systemd service, I get the following error:
```
error: builder for '/nix/store/4pnwpc3il1ni4isck7yf55k2gv8v69k7-unit-script-outline-start.drv' failed with exit code 1;
       last 17 log lines:
       >
       > In /nix/store/f9xrdza98q0xhhmcqslynfnd0r3xsjym-unit-script-outline-start/bin/outline-start line 6:
       > export SECRET_KEY="$(head -n1 /var/lib/outline/secret_key)"
       >        ^--------^ SC2155 (warning): Declare and assign separately to avoid masking return values.
       >
       >
       > In /nix/store/f9xrdza98q0xhhmcqslynfnd0r3xsjym-unit-script-outline-start/bin/outline-start line 7:
       > export UTILS_SECRET="$(head -n1 /var/lib/outline/utils_secret)"
       >        ^----------^ SC2155 (warning): Declare and assign separately to avoid masking return values.
       >
       >
       > In /nix/store/f9xrdza98q0xhhmcqslynfnd0r3xsjym-unit-script-outline-start/bin/outline-start line 12:
       > export OIDC_CLIENT_SECRET="$(head -n1 /run/secrets/outline/gitlab-auth-secret)"
       >        ^----------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.
       >
       > For more information:
       >   https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
       For full logs, run:
         nix log /nix/store/4pnwpc3il1ni4isck7yf55k2gv8v69k7-unit-script-outline-start.drv
```

It seems like making all shell scripts shellcheck compliant is an ongoing effort (see the description of `systemd.enableStrictShellChecks`). I want to do my part here :)

### Steps to reproduce

Enable `systemd.services.outline.enableStrictShellChecks` or `systemd.enableStrictShellChecks` with any normal outline configuration.

### Expected behaviour

The systemd service's `script` should be shellcheck compliant.

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.56, NixOS, 25.05 (Warbler), 25.05.20251110.c244830`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - nixpkgs: `/nix/store/y49qqnpqd04nkkanxrq1x637qa61ckpg-source`

### Notify maintainers

@cab404 @yrd (The module doesn't seem to have a dedicated maintainer)

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/nixos/modules/services/web-apps/outline.nix
+++ b/nixos/modules/services/web-apps/outline.nix
@@ -788,48 +788,62 @@ in
         '';
 
         script = ''
-          export SECRET_KEY="$(head -n1 ${lib.escapeShellArg cfg.secretKeyFile})"
-          export UTILS_SECRET="$(head -n1 ${lib.escapeShellArg cfg.utilsSecretFile})"
+          SECRET_KEY="$(head -n1 ${lib.escapeShellArg cfg.secretKeyFile})"
+          export SECRET_KEY
+          UTILS_SECRET="$(head -n1 ${lib.escapeShellArg cfg.utilsSecretFile})"
+          export UTILS_SECRET
           ${lib.optionalString (cfg.storage.storageType == "s3") ''
-            export AWS_SECRET_ACCESS_KEY="$(head -n1 ${lib.escapeShellArg cfg.storage.secretKeyFile})"
+            AWS_SECRET_ACCESS_KEY="$(head -n1 ${lib.escapeShellArg cfg.storage.secretKeyFile})"
+            export AWS_SECRET_ACCESS_KEY
           ''}
           ${lib.optionalString (cfg.slackAuthentication != null) ''
-            export SLACK_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.slackAuthentication.secretFile})"
+            SLACK_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.slackAuthentication.secretFile})"
+            export SLACK_CLIENT_SECRET
           ''}
           ${lib.optionalString (cfg.googleAuthentication != null) ''
-            export GOOGLE_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.googleAuthentication.clientSecretFile})"
+            GOOGLE_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.googleAuthentication.clientSecretFile})"
+            export GOOGLE_CLIENT_SECRET
           ''}
           ${lib.optionalString (cfg.azureAuthentication != null) ''
-            export AZURE_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.azureAuthentication.clientSecretFile})"
+            AZURE_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.azureAuthentication.clientSecretFile})"
+            export AZURE_CLIENT_SECRET
           ''}
           ${lib.optionalString (cfg.oidcAuthentication != null) ''
-            export OIDC_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.oidcAuthentication.clientSecretFile})"
+            OIDC_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.oidcAuthentication.clientSecretFile})"
+            export OIDC_CLIENT_SECRET
           ''}
           ${lib.optionalString (cfg.discordAuthentication != null) ''
-            export DISCORD_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.discordAuthentication.clientSecretFile})"
+            DISCORD_CLIENT_SECRET="$(head -n1 ${lib.escapeShellArg cfg.discordAuthentication.clientSecretFile})"
+            export DISCORD_CLIENT_SECRET
           ''}
           ${lib.optionalString (cfg.sslKeyFile != null) ''
-            export SSL_KEY="$(head -n1 ${lib.escapeShellArg cfg.sslKeyFile})"
+            SSL_KEY="$(head -n1 ${lib.escapeShellArg cfg.sslKeyFile})"
+            export SSL_KEY
           ''}
           ${lib.optionalString (cfg.sslCertFile != null) ''
-            export SSL_CERT="$(head -n1 ${lib.escapeShellArg cfg.sslCertFile})"
+            SSL_CERT="$(head -n1 ${lib.escapeShellArg cfg.sslCertFile})"
+            export SSL_CERT
           ''}
           ${lib.optionalString (cfg.slackIntegration != null) ''
-            export SLACK_VERIFICATION_TOKEN="$(head -n1 ${lib.escapeShellArg cfg.slackIntegration.verificationTokenFile})"
+            SLACK_VERIFICATION_TOKEN="$(head -n1 ${lib.escapeShellArg cfg.slackIntegration.verificationTokenFile})"
+            export SLACK_VERIFICATION_TOKEN
           ''}
           ${lib.optionalString (cfg.smtp != null) ''
-            export SMTP_PASSWORD="$(head -n1 ${lib.escapeShellArg cfg.smtp.passwordFile})"
+            SMTP_PASSWORD="$(head -n1 ${lib.escapeShellArg cfg.smtp.passwordFile})"
+            export SMTP_PASSWORD
           ''}
 
           ${
             if (cfg.databaseUrl == "local") then
               ''
-                export DATABASE_URL=${lib.escapeShellArg localPostgresqlUrl}
+                DATABASE_URL=${lib.escapeShellArg localPostgresqlUrl}
+                export DATABASE_URL
                 export PGSSLMODE=disable
               ''
             else
               ''
-                export DATABASE_URL=${lib.escapeShellArg cfg.databaseUrl}
+                DATABASE_URL=${lib.escapeShellArg cfg.databaseUrl}
+                export DATABASE_URL
               ''
           }
 

